PWGCF/MultiparticleCorrelations/MuPa-MemberFunctions.h

```diff
@@ -0,0 +1,362 @@
+{
+
+  // Copy histogram holding weights from an external file to the corresponding data member.
+
+  // Basic protection:
+  if (!(TString(variable).EqualTo("phi") || TString(variable).EqualTo("pt") || TString(variable).EqualTo("eta"))) {
+    cout << __LINE__ << endl;
+    exit(1);
+  }
+
+  Int_t ppe = -1;
+  if (TString(variable).EqualTo("phi")) {
+    ppe = 0;
+  }
+  if (TString(variable).EqualTo("pt")) {
+    ppe = 1;
+  }
+  if (TString(variable).EqualTo("eta")) {
+    ppe = 2;
+  }
+
+  // Finally:
+  hist->SetDirectory(0);
+  pw_a.fWeightsHist[ppe] = (TH1D*)hist->Clone(); // use eventually this line
+  //fWeightsHist = (TH1D*)hist->Clone();
+  if (!pw_a.fWeightsHist[ppe]) {
+    cout << __LINE__ << endl;
+    exit(1);
+  } // use eventually this line
+
+  // Flag:
+  // fUseWeights[ppe] = kTRUE; // use eventually this line
+
+} // void SetWeightsHist(TH1D* const hist, const char *variable)
+
+//============================================================
+
+TH1D* GetWeightsHist(const char* variable)
+{
+
+  // The standard getter.
+
+  // Basic protection:
+  if (!(TString(variable).EqualTo("phi") || TString(variable).EqualTo("pt") || TString(variable).EqualTo("eta"))) {
+    cout << __LINE__ << endl;
+    exit(1);
+  }
+
+  Int_t ppe = -1;
+  if (TString(variable).EqualTo("phi")) {
+    ppe = 0;
+  }
+  if (TString(variable).EqualTo("pt")) {
+    ppe = 1;
+  }
+  if (TString(variable).EqualTo("eta")) {
+    ppe = 2;
+  }
+
+  // Finally:
+  return pw_a.fWeightsHist[ppe]; // use eventually this line
+                                 //return fWeightsHist;
+
+} // TH1D* GetWeightsHist(const char *variable)
+
+//============================================================
+
+TH1D* GetHistogramWithWeights(const char* filePath, const char* variable)
+{
+  // ...
+
+  // a) Return value;
+  // b) Basic protection for arguments;
+  // c) Check if the external ROOT file exists at specified path;
+  // d) Access the external ROOT file and fetch the desired histogram with weights;
+  // e) Close the external ROOT file.
+
+  // a) Return value:
+  TH1D* hist = NULL;
+
+  // b) Basic protection for arguments:
+  if (!(TString(variable).EqualTo("phi") || TString(variable).EqualTo("pt") || TString(variable).EqualTo("eta"))) {
+    cout << __LINE__ << endl;
+    exit(1);
+  }
+
+  // c) Check if the external ROOT file exists at specified path:
+  /*
+ if(gSystem->AccessPathName(filePath,kFileExists))
+ {
+  //Red(Form("if(gSystem->AccessPathName(filePath,kFileExists)), filePath = %s",filePath)); // use eventually
+  cout<<__LINE__<<endl;
+  exit(1);
+ }
+ */
+
+  // d) Access the external ROOT file and fetch the desired histogram with weights:
+  //TFile *weightsFile = TFile::Open(filePath,"READ");
+  TGrid* alien = TGrid::Connect("alien", gSystem->Getenv("USER"), "", "");
+  if (!alien) {
+    cout << __LINE__ << endl;
+    exit(1);
+  }
+  TFile* weightsFile = TFile::Open(Form("alien://%s", filePath), "READ");
+
```

Comment:
I know how it was it in AliPhysics but now there is no `analysis.root` file any longer
The access to `alien` from the execution nodes is probably prohibited for performance reasons while the `CCDB` is replicated in the execution nodes 
I have not tried my self but I guess it will not work 
It was decided from the beginning that the corrections will be stored in `CCDB`

Reply:
OK, thanks for the further feedback! It suffices for this approach to work also in O2 to allow access to custom files in AliEn only in `init(...)`, but access to custom files clearly has to be disabled in `process(...)`, because that would be a massive performance penalty.

Reply:
In `init` you will have the same impact
Everything is run on the execution nodes, `init` and `process`
What I'm not sure is if you will get an execution error if you try to access an `alien` file or if your workflow will just hung forever

Reply:
> Everything is run on the execution nodes, init and process

Yes, but unlike ``process`, the `init` is executed only once, that was my point.

Reply:
Thanks for the pull request approval, and for taking the close look at my code!